src/tasks/block.rs

```diff
@@ -1,329 +1,352 @@
     }
 
-    async fn filter_transactions(&self, in_progress: &mut InProgressBlock) {
-        // query the rollup node to see which transaction(s) have been included
-        let mut confirmed_transactions = Vec::new();
-        for transaction in in_progress.transactions.iter() {
-            let tx = self
-                .ru_provider
-                .get_transaction_by_hash(*transaction.tx_hash())
-                .await
-                .expect("failed to get receipt");
-            if tx.is_some() {
-                confirmed_transactions.push(transaction.clone());
+    /// Continuously runs the block simulation and submission loop.
+    ///
+    /// This function clones the simulation cache, calculates a deadline for block building,
+    /// attempts to build a block using the latest cache and constants, and submits the built
+    /// block through the provided channel. If an error occurs during block building or submission,
+    /// it logs the error and continues the loop.
+    ///
+    /// This function runs indefinitely and never returns.
+    ///
+    /// # Arguments
+    /// - `constants`: The system constants for the rollup.
+    /// - `cache`: The simulation cache containing transactions and bundles.
+    /// - `submit_sender`: A channel sender used to submit built blocks.
+    async fn run_simulator(
+        self: Arc<Self>,
+        constants: SignetSystemConstants,
+        cache: SimCache,
+        submit_sender: mpsc::UnboundedSender<BuiltBlock>,
+    ) {
+        loop {
+            let sim_cache = cache.clone();
+            let finish_by = self.calculate_deadline();
+
+            match self.handle_build(constants, sim_cache, finish_by).await {
+                Ok(block) => {
+                    tracing::debug!(block = ?block, "built block");
+                    let _ = submit_sender.send(block);
+                }
+                Err(e) => {
+                    tracing::error!(err = %e, "failed to send block");
+                    continue;
+                }
             }
         }
-        trace!(confirmed = confirmed_transactions.len(), "found confirmed transactions");
-
-        // remove already-confirmed transactions
-        for transaction in confirmed_transactions {
-            in_progress.remove_tx(&transaction);
-        }
     }
 
-    // calculate the duration in seconds until the beginning of the next block slot.
-    fn secs_to_next_slot(&self) -> u64 {
-        let curr_timestamp: u64 = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
-        let current_slot_time = (curr_timestamp - self.config.chain_offset) % ETHEREUM_SLOT_TIME;
-        (ETHEREUM_SLOT_TIME - current_slot_time) % ETHEREUM_SLOT_TIME
-    }
+    /// Calculates the deadline for the current block simulation.
+    ///
+    /// # Returns
+    ///
+    /// An `Instant` representing the deadline.
+    pub fn calculate_deadline(&self) -> Instant {
+        let now = SystemTime::now();
```

Comment:
i don't like that we're constantly mixing timestamps and Duration/Instant. it makes it very hard to reason about the logic. cc @anna-carroll can we add more convenience functions to `SlotCalculator` to make these easier? what are the common actions we want to take using a calculator?

Reply:
Created [ENG-1019](https://linear.app/initiates/issue/ENG-1019/standardize-instant-and-duration-time-usage-in-slot-calculator) to capture this.